### PR TITLE
SB.config bug fixes

### PR DESCRIPTION
**Pull request by OfficialNoob**
This should fix:

- Config values not getting decoded from JSON on page load

### CODE REVIEW

File: content.js [hunk 1]
```diff
@@ -987,11 +987,9 @@ function sendSubmitMessage(){
...
                 submitButton.addEventListener("animationend", animationEndListener);
 
-                //clear the sponsor times
-                SB.config.sponsorTimes.delete(currentVideoID);
-
                 //add submissions to current sponsors list
-                sponsorTimes = sponsorTimes.concat(sponsorTimesSubmitting);
+                SB.config.sponsorTimes.set(currentVideoID, sponsorTimesSubmitting);
+                
                 for (let i = 0; i < sponsorTimesSubmitting.length; i++) {
```
Comment (ajayyy) [inline content.js]: This is when the times are cleared after a submit. Why are you adding them back?

File: SB.js [hunk 2]
```diff
@@ -184,8 +184,8 @@ function resetConfig() {
     SB.config = SB.defaults;
 };
 
-function convertJSON() {
-    Object.keys(SB.defaults).forEach(key => {
+function decodeConfig() {
+    Object.keys(SB.localConfig).forEach(key => {
```
Comment (ajayyy) [inline SB.js]: Oh shoot, how was this ever working?

Comment (OfficialNoob) [reply SB.js]: If the key was used in SB.defaults it would still get decoded

File: content.js (hunk 1 above)
Comment (OfficialNoob) [reply content.js]: Reverted

File: SB.js (hunk 2 above)
Comment (ajayyy) [reply SB.js]: Isn't the sponsorTimes in the defaults? Doesn't that mean that this shouldn't change the behavior?

**Comment by ajayyy** [comment]
Is there still an issue related to sponsorTimes.concat?

**Comment by OfficialNoob** [comment]
There was but it may have been because SB.config.sponsorTimes was not working

**Comment by ajayyy** [comment]
https://github.com/ajayyy/SponsorBlock/issues/258

File: SB.js (hunk 2 above)
Comment (OfficialNoob) [reply SB.js]: It is in the defaults however there maybe something thats not.

Comment (OfficialNoob) [reply SB.js]: I have been unable to get the error after this change maybe it only happens when the video has no times

**Comment by OfficialNoob** [comment]
There could be an issue if a string was set as valid JSON

**Comment by OfficialNoob** [comment]
@ajayyy should there be a type change prevention on the proxy?

This would: 
- allow custom objects or Maps to be detected easier and more reliable
- make config variables more predictable

The type could still be changed with SB.config.types.set("prop", "array")
```
function getName(obj) {
    return Object.prototype.toString.call(obj).replace(/^\[object (.+)\]$/,"$1").toLowerCase(); 
}
```

**Comment by ajayyy** [comment]
@OfficialNoob yea, that's a great idea. But, there has to be a better way to do it then that.

**Comment by ajayyy** [comment]
Also, is there a reason that the array is stored as a string? Chrome supports storing arrays

**Comment by OfficialNoob** [comment]
@ajayyy because I dont want arrays to be converted to Maps
This would not be needed if I could get the type using `SB.config.types`

**Comment by OfficialNoob** [comment]
`SB.config.types` would contain the initiation variable type such as from `SB.defaults`
Unless changed

**Comment by ajayyy** [comment]
Check the typescript branch. Can we use typescript types to achieve this?

**Pull request closed without merging**